photon-lib/src/main/scala/com/linkedin/photon/ml/util/VectorUtils.scala

```diff
@@ -206,4 +206,22 @@ object VectorUtils {
     v1.length == v2.length && v1.toArray.zip(v2.toArray).forall { case (m1, m2) =>
       MathUtils.isAlmostZero(m2 - m1)
     }
+
+  /**
+   * Returns the indices for non-zero elements of the vector
+   *
+   * @param vector the input vector
+   * @return the set of indices
+   */
+  def getActiveIndices(vector: Vector[Double]): Set[Int] = vector match {
+    case vector: DenseVector[Double] => vector
```

Comment:
Isn't a one pass declarative loop more efficient?

Reply:
Sorry, not following you here. Can you elaborate?

Reply:
A declarative loop can find the actives indices in 1 single iteration over the vector. How many iterations over the vector are performed by this code?

Reply:
Ah, good point. Will fix.

Reply:
@fastier-li done.